src/extensions/step_handlers/script_handler/src/script_handler.cpp

```diff
@@ -501,19 +503,38 @@ ADUC_Result ScriptHandlerImpl::PrepareScriptArguments(
  */
 ADUC_Result ScriptHandlerImpl::Install(const tagADUC_WorkflowData* workflowData)
 {
-    ADUC_Result result = PerformAction("--action-install", workflowData);
+    ADUC_Result result = PerformAction("install", workflowData);
     return result;
 }
 
-static ADUC_Result ScriptHandler_PerformAction(const std::string& action, const tagADUC_WorkflowData* workflowData)
+/**
+ * @brief Perform a workflow action. If @p prepareArgsOnly is true, only prepare data, but not actually
+ *        perform any action.
+ *
+ * @param action Indicate an action to perform. This can be 'download', 'install',
+ *               'apply', "cancel", and "is-installed".
+ * @param workflowData An object containing workflow data.
+ * @param prepareArgsOnly  Boolean indicates whether to prepare action data only.
+ * @param[out] scriptFilePath Output string contains a script to be run.
+ * @param[in] args List of options and arguments.
+ * @param[out] commandLineArgs An output command-line arguments.
+ * @param[out] scriptOutput If @p prepareArgsOnly is false, this will contains the action output string.
+ * @return ADUC_Result
+ */
+ADUC_Result ScriptHandler_PerformAction(
+    const std::string& action,
+    const tagADUC_WorkflowData* workflowData,
+    bool prepareArgsOnly,
+    std::string& scriptFilePath,
```

Comment:
bunch of non-const references?  are these out parameters?  If so, this would be much better as a struct/class that's returned, e.g.

ActionValues {
  string args;
  string output;
  ADUC_Result result;
}

ActionValues PerformAction()

Reply:
Changed. Thanks for the suggestion.